### DISCUSSION TITLE
New Dip Buying Mechanics?

### DISCUSSION THREAD

**Discussion by rhettre** [TheTallMan67/Gemini-API-Functions-for-AWS#11] [Ideas]
A guy in the comments came up with new use case:

He's posting limit orders that are 10% lower than spot price every week. If the order doesn't fill in the next week he wants the order to get cancelled and to be re-placed at 10% lower than the current spot price - because he doesn't want lots of orders on the books tying up his liquidity. 

Looks like there's cancel order function and cancel all orders function available in API - cancel single order may be easier for this use case (by looking at timestamp on order).

Think we should be able to get order status timestamp field to see when it was posted and if X days have passed cancel the order and make a new one. 

Or bulk cancel all orders and just run that script every X days

**Comment by TheTallMan67** [comment]
you could also consider writing the order Ids to dynamoDB (or spreadsheet) so they can be cancelled by ID later

**Reply by manofthemountain** [reply]
I think you would use the 'Get Active Orders' post and then compare the price of the order to your threshold (10%) from the spot price, this would then let you know if you want to canx the order or leave it open.
